actors/cloudbroker/cloudbroker__computenode/methodclass/cloudbroker_computenode.py

```diff
@@ -150,27 +140,33 @@ def maintenance(self, id, gid, vmaction, **kwargs):
         """
         if vmaction not in ('move', 'stop'):
             raise exceptions.BadRequest("VMAction should either be move or stop")
-        stack = self._getStack(id, gid)
+
+        machines_actor = j.apps.cloudbroker.machine
+
+        stack = self._getStack(id)
         errorcb = functools.partial(self._errorcb, stack)
         self._changeStackStatus(stack, "MAINTENANCE")
-        title = 'Putting Node in Maintenance'
+
+        if self.models.Cloudspace.objects(stack=stack, status="DEPLOYED").count() > 0 and vmaction == 'move':
+            raise exceptions.BadRequest("Can not put stack in maintenance when it has VFW deployed")
+
         if vmaction == 'stop':
-            machines_actor = j.apps.cloudbroker.machine
-            stackmachines = self._get_stack_machines(stack['id'], ['id', 'status', 'tags'])
+            stackmachines = self.models.VMachine.objects(stack=stack)
```

Comment:
please use the _get_stakc_machiens method instead to make sure you use the correct filtering